src/edu/washington/escience/myria/api/encoding/IDBControllerEncoding.java

```diff
@@ -34,8 +36,13 @@
 
   @Override
   public IDBController construct(final ConstructArgs args) {
-    return new IDBController(argSelfIdbId, realEosControllerOperatorId, realEosControllerWorkerId, null, null, null,
-        argState.construct(), MoreObjects.firstNonNull(sync, Boolean.FALSE));
+    IDBController controller =
+        new IDBController(argSelfIdbId, realEosControllerOperatorId, realEosControllerWorkerId, null, null, null,
+            argState.construct(), MoreObjects.firstNonNull(sync, Boolean.FALSE));
+    if (relationKey != null) {
+      controller.setStoreRelationKey(relationKey);
```

Comment:
Because in some cases we don't need to store the states to disk, i.e. there are multiple IDBs in a query but only one serves as the final query result.